src/doi/__init__.py

```diff
@@ -79,12 +74,12 @@ def validate_doi(doi):
         raise ValueError('Something unexpected happened')
```

Comment:
There's still a typing error with this if-else chain because the first if does not return anything. What should it return?

Reply:
First of all thank you very much for this amazing work, wonderful stuff going on here.
After the typing signature, this function if everything goes well should just return `None`,
otherwise il will raise an exception.
Actually the other exceptions seem to be overlaping, anyways the only thing that we
care about in this function is that the doi is actually valid.
So the exceptions at the end do not actually make a lot of sense, I'll remove them.
Otherwise I think the code is very good and it passes the `mypy --strict` test,
I'll push it then.

Thanks!